src/forms/Schedule.jsx

```diff
@@ -47,11 +47,32 @@ const fetchClasses = async (term, course) => {
   return componentToClasses;
 };
 
+export const fetchCourseByTerm = async (term) => {
+  try {
+    const response = await fetch(`${API_URL}/api/v1/courses/?term=${term}`);
+    const data = await response.json();
+
+    const sortedCoursesAvailable = await sortObj(data.objects);
+    return sortedCoursesAvailable;
+  } catch (error) {
+    console.log(`Error fetching terms: ${error}`);
+    return [];
+  }
+};
+
 export const Form = (props) => {
   const { values, handleChange, setValues } = useFormContext();
-  const [courseOptions, setCourseOptions] = useState([]);
+  const [courseOptions, setCourseOptions] = useState(null);
   const [showAdvanced, setShowAdvanced] = useState(false);
-  const [componentData, setComponentData] = useState({});
+  const [componentData, setComponentData] = useState(props.courseData);
+
+  if (courseOptions === null) {
+    setCourseOptions([]);
+    fetchCourseByTerm(props.term)
+      .then((data) => setCourseOptions(data))
+      .catch((err) => console.log(`Error fetching course options: ${err}`));
+  }
+
```

Comment:
Put this in a useEffect or handler, setting state like this is gonna cause this to execute on each re-render